android-components/components/feature/fxsuggest/src/main/java/mozilla/components/facts/FxSuggestFacts.kt

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.fxsuggest.facts
+
+import mozilla.components.feature.fxsuggest.FxSuggestClickInfo
+import mozilla.components.support.base.Component
+import mozilla.components.support.base.facts.Action
+import mozilla.components.support.base.facts.Fact
+import mozilla.components.support.base.facts.collect
+
+/**
+ * Facts emitted for telemetry related to the Firefox Suggest feature.
+ */
+class FxSuggestFacts {
+    /**
+     * Specific types of telemetry items.
+     */
+    object Items {
+        const val SPONSORED_SUGGESTION_CLICKED = "sponsored_suggestion_clicked"
+    }
+}
+
+private fun emitFxSuggestFact(
+    action: Action,
+    item: String,
+    value: String? = null,
+    metadata: Map<String, Any>? = null,
+) {
+    Fact(
+        Component.FEATURE_FXSUGGEST,
+        action,
+        item,
+        value,
+        metadata,
+    ).collect()
+}
+
+internal fun emitSponsoredSuggestionClickedFact(clickInfo: FxSuggestClickInfo) {
+    emitFxSuggestFact(
+        Action.INTERACTION,
+        FxSuggestFacts.Items.SPONSORED_SUGGESTION_CLICKED,
+        metadata = mapOf("clickInfo" to clickInfo),
```

Comment:
Suggestion: Let's declare a new object on `FxSuggestFacts` called `MetadataKeys`, and put this as a constant there, [like this](https://searchfox.org/mozilla-mobile/rev/422a641552f7fcb3ed0dbcd9e3dc2c9034280cd1/firefox-android/android-components/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBarFacts.kt#23-28)—so that we can refer to it as `FxSuggestFacts.MetadataKeys.CLICK_INFO` here and in `MetricController`, instead of repeating the string.